asyncstatsd/metrics/basic.py

```diff
@@ -0,0 +1,61 @@
+from dataclasses import dataclass, field
+from datetime import timedelta
+
+
+@dataclass
+class StatsdMetric:
+    name: str
+    value: int | float | timedelta
+    unit: str
+    rate: float = field(default=1.0)
+
+    def get_value(self) -> str:
+        return f"{self.value}"
+
+    def serialize(self) -> str | None:
+        value = self.get_value()
+        serialized = f"{self.name}:{value}|{self.unit}"
+        if self.rate < 1:
+            serialized = f"{serialized}|@{self.rate}"
+        return serialized
+
+
+@dataclass
+class CounterMetric(StatsdMetric):
+    unit: str = "c"
+    value: int = 1
+
+
+class GaugeMetricMixin:
+    def get_value(self) -> str:
+        value = self.value  # type: ignore
+        if self.delta:  # type: ignore
+            sign = "+" if value > 0 else ""
+            value = f"{sign}{value}"
```

Comment:
deltas with value 0 won't get a sign, is that expected?
I'm not sure, but looking at the test `test_serialize_delta_full_rate()` it seems to me the presence of the sign distinguishes a delta value from a non delta value in the serialisation